docs/DeveloperGuide.md

```diff
@@ -1,11 +1,308 @@
 # Developer Guide
 
-## Design & implementation
+## Table of Contents
+1. [Introduction](#introduction)
+    1. [Background](#background)
+    1. [Purpose](#purpose)
+    1. [Scope](#scope)
+1. [Getting Started](#getting-started)
+    1. [Prerequisites](#prerequisites)
+    1. [Setting Up](#setting-up)
+    1. [Running the Program](#running-the-program)
+1. [Design](#design)
+    1. [Architecture](#architecture)
+    1. [UI Component](#ui-component)
+    1. [Logic Component](#logic-component)
+    1. [Model Component](#model-component)
+    1. [Storage Component](#storage-component)
+1. [Implementation](#implementation)
+    1. [Project](#project)
+        1. [Create Project](#create-project)
+        1. [Select Project](#select-project)
+    1. [Task](#task)
+        1. [Add Task](#add-task)
+        1. [View Task](#view-task)
+        1. [Delete Task](#delete-task)
+        1. [Change Task Priority](#change-task-priority)
+        1. [Mark Task as Complete](#mark-task-as-complete)
+    1. [Sprint](#sprint)
+        1. [Create Sprint](#create-sprint)
+        1. [View Sprint](#view-sprint)
+        1. [Add Task to Sprint](#add-task-to-sprint)
+        1. [Remove Task from Sprint](#remove-task-from-sprint)
+        1. [Allocate Sprint Tasks to Members](#allocate-sprint-tasks-to-members)        
+    1. [Storage](#storage)
+        1. [Location](#location)
+        1. [Loading Data](#loading-data)
+            1. [Converting and Mapping of JSON to Objects](#converting-and-mapping-of-json-to-objects)
+        1. [Saving Data](#saving-data)
+            1. [When the Program Exits](#when-the-program-exits)
+            1. [Changes Made to the Data](#changes-made-to-the-data)
+            1. [Serialising Objects to JSON](#serialising-objects-to-json)
+1. [Appendix: Requirements]()
+1. [Others](#target-user-profile)
```

Comment:
I feel that it might be better to use numbering like 1.1, 1.2 so that the user would be able to see the hierarchy better